### PR TITLE
Revert "Upgrade Gradle and AGP versions to 7.5/7.2 and migrate examples/tests"

### PR DESCRIPTION
**Pull request by zanderso**
Reverts flutter/flutter#108197

Failing consistently on Mac_android microbenchmarks:
```
[microbenchmarks] [STDOUT] [STDOUT] [+1142 ms] > Configure project :
[microbenchmarks] [STDOUT] [STDOUT] [        ] Evaluating root project 'android' using build file '/opt/s/w/ir/x/w/recipe_cleanup/tmp17zz_ptv/flutter sdk/dev/benchmarks/microbenchmarks/android/build.gradle'.
[microbenchmarks] [STDOUT] [STDOUT] [        ] Resource missing. [HTTP GET: https://dl.google.com/dl/android/maven2/com/android/tools/build/gradle/%7B%7BagpVersion%7D%7D/gradle-%7B%7BagpVersion%7D%7D.pom]
[microbenchmarks] [STDOUT] [STDOUT] [        ] Resource missing. [HTTP GET: https://repo.maven.apache.org/maven2/com/android/tools/build/gradle/%7B%7BagpVersion%7D%7D/gradle-%7B%7BagpVersion%7D%7D.pom]
[microbenchmarks] [STDOUT] [STDOUT] [+2222 ms] Running Gradle task 'assembleProfile'... (completed in 6.6s)
[microbenchmarks] [STDOUT] [STDOUT] [   +3 ms] "flutter run" took 7,347ms.
[microbenchmarks] [STDOUT] [STDOUT] [   +1 ms] ensureAnalyticsSent: 0ms
[microbenchmarks] [STDOUT] [STDOUT] [        ] Running shutdown hooks
[microbenchmarks] [STDOUT] [STDOUT] [        ] Shutdown hooks complete
[microbenchmarks] [STDOUT] [STDOUT] [        ] exiting with code 1
```

### CODE REVIEW

**Review by GaryQian — APPROVED** [review]

**Comment by zanderso** [comment]
> * Please get at least one approved review if you are already a member or two member reviews if you are not a member before re-applying this label. **Reviewers**: If you left a comment approving, please use the "approve" review action instead.

@keyonghan @godofredoc  Team members should be able to land reverts to green the tree without waiting for approval.

**Comment by godofredoc** [comment]
> > * Please get at least one approved review if you are already a member or two member reviews if you are not a member before re-applying this label. **Reviewers**: If you left a comment approving, please use the "approve" review action instead.
> 
> @keyonghan @godofredoc Team members should be able to land reverts to green the tree without waiting for approval.

@ricardoamador is working on a revert api that will allow to implement that functionality. Unfortunately github does not provide any api to differentiate regular PRs from reverts.

**Pull request merged**
